src/TSHelper.ts

```diff
@@ -652,6 +652,11 @@ export class TSHelper {
         return match && match[0] === str;
     }
 
+    public static isValidLuaFunctionDeclarationName(str: string): boolean {
```

Comment:
What is the difference between this and `isValidLuaIdentifier`?

Reply:
This allows a period `.`

Reply:
This might be a little nitpicky, but would it make sense/ be worth it to implement this function as `str.split(".").every(s => s.length > 0 && isValidLuaIdentifier(s))`?

Reply:
To be honest, I'm not sure that's any clearer. Perhaps just a comment explaining the difference would be suffice?

Reply:
That works for me.